sway-core/src/type_system/unify/unifier.rs

```diff
@@ -137,6 +137,8 @@ impl<'a> Unifier<'a> {
                 )
             }
 
+            (Never, Never) => (),
```

Comment:
Why this special case here? Isn't that case also covered with the below cases with `Never` and placeholder?